cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1800,7 +1859,8 @@ public List<Condition> getWarningConditions() {
      *          broker nodes when KRaft is enabled.
      */
     private Labels brokersSelector()    {
-        if (useKRaft)   {
+        // Starting from the migration phase, brokers should be already selected and used via KRaft
+        if (this.kafkaMetadataConfigState.ordinal() >= KafkaMetadataStateManager.KafkaMetadataConfigurationState.MIGRATION.ordinal()) {
```

Comment:
FYI: This looks like the methods in `KafkaBrokerConfigurationBuilder` I suggested should be instance methods instead. It seems they would have a lot of value also in this class.

Reply:
Yeah, I will make the changes.